framework/src/base/FEProblem.C

```diff
@@ -1882,6 +1898,31 @@ FEProblem::reinitMaterials(SubdomainID blk_id, THREAD_ID tid, bool swap_stateful
 }
 
 void
+FEProblem::reinitMaterialsDirac(SubdomainID blk_id, THREAD_ID tid, bool swap_stateful)
+{
+  if (_all_materials[Moose::DIRAC_MATERIAL_DATA].hasActiveBlockObjects(blk_id, tid))
+  {
+    const Elem * & elem = _assembly[tid]->elem();
+    unsigned int n_points = _assembly[tid]->qRule()->n_points();
+
+    if (_dirac_material_data[tid]->nQPoints() != n_points)
```

Comment:
What happens if the number of points in the elem stays the same... but they MOVE within the element?  The stateful data is no longer valid... but this PR will still happily return old values...